plugins/cpp_metrics/service/cxxmetrics.thrift

```diff
@@ -4,26 +4,72 @@ include "project/project.thrift"
 namespace cpp cc.service.cppmetrics
 namespace java cc.service.cppmetrics
 
-enum CppMetricsType
+enum CppUnitType
+{
+  AstNodeUnit = 1,
+  FileUnit = 2
+}
+
+enum CppAstNodeMetricsType
 {
   ParameterCount = 1,
   McCabe = 2,
   LackOfCohesion = 3,
   LackOfCohesionHS = 4
 }
 
-struct CppMetricsTypeName
+enum CppModuleMetricsType
+{
+  Placeholder = 1
+}
+
+struct CppAstNodeMetricsTypeName
+{
+  1:CppAstNodeMetricsType type,
+  2:string name
+}
+
+struct CppModuleMetricsTypeName
 {
-  1:CppMetricsType type,
+  1:CppModuleMetricsType type,
   2:string name
 }
 
 struct CppMetricsAstNode
 {
-  1:CppMetricsType type,
+  1:CppAstNodeMetricsType type,
   2:double value
 }
 
+struct CppAllMetricsAstNode
```

Comment:
Putting `All` inbetween `Cpp` and `Metrics`, while the module's name is `cppmetrics` is highly odd, I suggest naming it one of the following:
- `AllCppMetricsAstNode`
- `CppMetricsAstNodeAll`
- `MultiCppMetricsAstNode`
- `CppMetricsAstNodeMulti`

Reply:
Fixed in e09e8ff80ab891cd8da090bdd74b5fe870ab6c1c.